src/overlays/actors/ovl_En_Mnk/z_en_mnk.c

```diff
@@ -28,191 +55,2186 @@ ActorInit En_Mnk_InitVars = {
+
+        default:
+            this->unk_3E0 = 0;
+            break;
+    }
+    this->animIndex = animIndex;
+}
+
+void EnMnk_TiedUpMonkey_SetAnim(EnMnk* this, s32 animIndex) {
+    EnMnk_TiedUpMonkey_ChangeAnim(this, animIndex, ANIMMODE_LOOP, -5.0f);
+}
+
+void EnMnk_TiedUpMonkey_SetNewAnim(EnMnk* this, s32 animIndex) {
+    if (this->animIndex != animIndex) {
+        EnMnk_TiedUpMonkey_ChangeAnim(this, animIndex, ANIMMODE_LOOP, -5.0f);
+    }
+}
+
+void EnMnk_HangingMonkey_ChangeAnim(EnMnk* this, s32 animIndex, u8 mode, f32 transitionRate) {
+    Animation_Change(&this->skelAnime, sHangingMonkeyAnimations[animIndex], 1.0f, 0.0f,
+                     Animation_GetLastFrame(&sHangingMonkeyAnimations[animIndex]->common), mode, transitionRate);
+    Animation_Change(&this->propSkelAnime, sHangingMonkeyRopeAnimations[animIndex], 1.0f, 0.0f,
+                     Animation_GetLastFrame(&sHangingMonkeyRopeAnimations[animIndex]->common), mode, transitionRate);
+    this->animIndex = animIndex;
+}
+
+void EnMnk_HangingMonkey_SetAnim(EnMnk* this, s32 animIndex) {
+    EnMnk_HangingMonkey_ChangeAnim(this, animIndex, ANIMMODE_LOOP, -5.0f);
+}
+
+void EnMnk_Monkey_ChangeAnim(EnMnk* this, s32 animIndex, u8 mode, f32 morphFrames) {
+    s32 pad;
+
+    if (animIndex == 100) {
+        animIndex = (s32)Rand_ZeroFloat(3.99f) + 7;
+    }
+    if ((animIndex == 4) || (animIndex == 6)) {
+        Animation_Change(&this->skelAnime, sMonkeyAnimations[animIndex], -1.0f,
+                         Animation_GetLastFrame(&sMonkeyAnimations[animIndex]->common), 0.0f, mode, morphFrames);
+    } else {
+        Animation_Change(&this->skelAnime, sMonkeyAnimations[animIndex], 1.0f, 0.0f,
+                         Animation_GetLastFrame(&sMonkeyAnimations[animIndex]->common), mode, morphFrames);
+    }
+    this->unk_3E0 = 0;
+    this->animIndex = animIndex;
+}
+
+void EnMnk_Monkey_SetAnim(EnMnk* this, s32 animIndex) {
+    EnMnk_Monkey_ChangeAnim(this, animIndex, ANIMMODE_LOOP, -5.0f);
+}
+
+void EnMnk_Monkey_SetupWaitToRunAndWaitAtEachPoint(EnMnk* this, PlayState* play) {
+    this->picto.actor.textId = 0x7D8;
+    this->actionFunc = EnMnk_Monkey_WaitToRunAndWaitAtEachPoint;
+    this->destPointIndex = 0;
+    this->unk_3E4 |= 0x3;
+    this->unk_3E4 &= ~0x40;
+    Animation_PlayLoop(&this->skelAnime, &object_mnk_Anim_008814);
+    this->picto.actor.velocity.y = 0.0f;
+    this->picto.actor.terminalVelocity = 0.0f;
+    this->picto.actor.gravity = 0.0f;
+}
+
+void EnMnk_Monkey_StartInvisible(EnMnk* this, PlayState* play) {
+    this->picto.actor.draw = NULL;
+    this->picto.actor.flags &= ~ACTOR_FLAG_1;
+    this->collider.dim.radius = 100;
+    this->unk_3E4 |= 0x8;
+    this->unk_3E4 |= 0x20;
+    if ((MONKEY_GET_SWITCHFLAG(this) != 0x7F) && Flags_GetSwitch(play, MONKEY_GET_SWITCHFLAG(this))) {
+        Actor_Kill(&this->picto.actor);
+        return;
+    }
+}
+
+void EnMnk_TiedUpMonkey_Init(EnMnk* this, PlayState* play) {
+    EnMnk* this2 = this;
```

Comment:
Instead of a `this2` does it work to create a `Actor* thisx = &this->actor` and use that wherever only actor variables are used. If not it might make sense to change these `Init` functions to take `Actor*` instead.

Reply:
It doesn't work unfortunately. So make it an `Actor*` arg instead?

Reply:
I would say yeah, does the other `Init` function have room for a cast (or 2) pad(s)

Reply:
It does have room yeah